src/adapters/mqtt/index.ts

```diff
@@ -42,12 +46,18 @@ class MqttAdapter extends Adapter {
       const certsConfig = process.env.GLEE_SERVER_CERTS?.split(',').map(t => t.split(':'))
       const certs = certsConfig?.filter(tuple => tuple[0] === this.serverName)?.map(t => fs.readFileSync(t[1])) // eslint-disable-line security/detect-non-literal-fs-filename
 
+      const serverBinding = mqttServerBinding
+      const protocolVersion = mqtt5ServerBinding ? 5 : 4
```

Comment:
You should get `protocolVersion` from the server object instead:

```suggestion
      const protocolVersion = parseInt(this.AsyncAPIServer.protocolVersion() || 4)
      const serverBinding = protocolVersion === 5 ? mqtt5ServerBinding : mqttServerBinding
```

Just because the `mqtt5` exists in the AsyncAPI document it doesn't mean it should be used.